src/components/LegacyPlanUpgradeModal/LegacyPlanUpgradeModal.tsx

```diff
@@ -0,0 +1,140 @@
+import { Box, Button, Dialog, Text } from '@/ui';
+import { Modal } from '../Modal/Modal';
+import { DividerElement } from '@/ui/Divider/Divider.styles';
+import { FleekLogo } from '../FleekLogo/FleekLogo';
+import { Link } from '../ftw/Link/Link';
+import { useEffect, useState } from 'react';
+import { useAuthContext } from '@/providers/AuthProvider';
+import { useFleekCheckout } from '@/hooks/useFleekCheckout';
+import { useToast } from '@/hooks/useToast';
+import { Icon } from '@/ui';
+import { useBillingContext } from '@/providers/BillingProvider';
+
+const PERKS = [
+  'Unlimited team members',
+  'Unlimited custom domains',
+  'Unlimited sites',
+  'Email support',
+] as const;
+
+// TODO add link to blog post later
+const LEARN_MORE_LINK = undefined;
+
+const SHOWN_KEY_PREFIX = 'legacy_plan_modal_shown_';
+
+export const LegacyPlanUpgradeModal = () => {
+  const [isOpen, setIsOpen] = useState(false);
+  const auth = useAuthContext();
+  const shownKey = `${SHOWN_KEY_PREFIX}${auth.accessToken}`;
```

Comment:
It's best to remove the accessToken; if using just the key is not enough, use the project id.

The accessToken contains the application state, which causes it to change on the project switch, etc. Although the access token shouldn't contain app state, it differ on every new token regardless if same project id; we'd likely be showing the warning several times to the end user, which might be too annoying.